misc/python/materialize/mzbuild.py

```diff
@@ -756,21 +761,47 @@ def __init__(self, dependencies: Iterable[Image]):
             image.acquired = image.spec() in known_images
             self._dependencies[d.name] = image
 
+    def cargo_common_build(self, deps_to_build: List[ResolvedImage]) -> None:
```

Comment:
Can you extract part of this function into a `@staticmethod` on `CargoBuild`? That will better encapsulate the logic. I think something like `CargoBuild.run_common_build(cargo_builds: list[CargoBuild])`, which takes a list of pre-images that are known to be of type `CargoBuild` and runs the common `cargo build` command that is necessary to build those images. Then this function can limit its scope to sniffing out `CargoBuild` preimages and invoking `CargoBuild.run_common_build` on those instances.